### PR TITLE
Conform to Navigation changes from OSD core

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Adopts to nav changes in core when a feature flag is enabled

### Category
Enhancement
### Why these changes are required?
Fix: #2020 

### What is the old behavior before changes and new behavior after changes?

https://github.com/user-attachments/assets/30c7dab9-2b7d-4006-a3bf-b8cad97b2e03

The first part of the video is the current behavior and the behavior with the feature flag off, the second part of the video is the new behavior with the feature flag on. 

### Issues Resolved
Fix: #2020
### Testing
Manual testing. Since the feature branch is not yet merged in core and many of the API calls are simply registration calls from exposed APIs code tests don't seem feasible. 
### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/plugin.ts [hunk 1]
```diff
@@ -97,11 +119,33 @@ export class SecurityPlugin
...
+        deps.dataSourceManagement
+      );
+    };
+
     if (mdsEnabled || apiPermission) {
       core.application.register({
         id: PLUGIN_NAME,
         title: 'Security',
         order: 9050,
+        workspaceAvailability: WorkspaceAvailability.outsideWorkspace,
```
Comment (SuZhou-Joe) [inline public/plugin.ts:148]:
```suggestion
        workspaceAvailability: WorkspaceAvailability.outsideWorkspace,
        NavLinkStatus: core.chrome.navGroup.AppNavLinkStatus.getNavGroupEnabled() ? AppNavLinkStatus.hidden : AppNavLinkStatus.visible,
```
Hi @derek-ho , we need to hide the old getting started entry from left navigation if the navGroup is enabled.

File: public/apps/configuration/app-router.tsx [hunk 2]
```diff
@@ -151,11 +151,10 @@ export interface DataSourceContextType {
   setDataSource: React.Dispatch<React.SetStateAction<DataSourceOption>>;
 }
 
-export const LocalCluster = { label: 'Local cluster', id: '' };
-
 export const DataSourceContext = createContext<DataSourceContextType | null>(null);
 
 export function AppRouter(props: AppDependencies) {
+  console.log(props.redirect);
```
Comment (SuZhou-Joe) [inline public/apps/configuration/app-router.tsx:157]: The log should be delete.

Comment (SuZhou-Joe) [inline public/apps/configuration/app-router.tsx:157]:
```suggestion
```
The log should be delete.

File: public/plugin.ts (hunk 1 above)
Comment (SuZhou-Joe) [inline public/plugin.ts:148]:
```suggestion
        workspaceAvailability: WorkspaceAvailability.outsideWorkspace,
        navLinkStatus: core.chrome.navGroup.getNavGroupEnabled() ? AppNavLinkStatus.hidden : AppNavLinkStatus.visible
```
The getting started entry will not be shown in new left navigation.

**Review by SuZhou-Joe — APPROVED** [review]

File: public/plugin.ts (hunk 1 above)
Comment (derek-ho) [reply public/plugin.ts:149]: Done

File: public/apps/configuration/app-router.tsx (hunk 2 above)
Comment (derek-ho) [reply public/apps/configuration/app-router.tsx]: Removed

**Review by scrawfor99 — APPROVED** [review]

**Review by DarshitChanpura — APPROVED** [review]

**Pull request merged**